packages/announcement-controller/src/AnnouncementController.ts

```diff
@@ -1,96 +1,116 @@
+  payload: [AnnouncementState, Patch[]];
+};
+
+const controllerName = 'AnnouncementController';
 
 const defaultState = {
   announcements: {},
 };
 
+const metadata = {
+  announcements: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+export type AnnouncementControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AnnouncementControllerActions,
+  AnnouncementControllerEvents,
+  never,
+  never
+>;
+
 /**
  * Controller for managing in-app announcements.
  */
-export class AnnouncementController extends BaseController<
-  AnnouncementConfig,
-  AnnouncementState
+export class AnnouncementController extends BaseControllerV2<
+  typeof controllerName,
+  AnnouncementState,
+  AnnouncementControllerMessenger
 > {
   /**
    * Creates a AnnouncementController instance.
    *
-   * @param config - Initial options used to configure this controller.
-   * @param state - Initial state to set on this controller.
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
    */
-  constructor(config: AnnouncementConfig, state?: AnnouncementState) {
-    super(config, state || defaultState);
-    this.initialize();
-    this._addAnnouncements();
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: AnnouncementControllerMessenger;
+    state?: AnnouncementState;
+  }) {
+    const mergedState = { ...defaultState, ...state };
+    super({ messenger, metadata, name: controllerName, state: mergedState });
+    this._addAnnouncements(mergedState.announcements);
   }
 
   /**
    * Compares the announcements in state with the announcements from file
    * to check if there are any new announcements
    * if yes, the new announcement will be added to the state with a flag indicating
    * that the announcement is not seen by the user.
+   *
+   * @param allAnnouncements - all announcements to
    */
-  private _addAnnouncements(): void {
+  private _addAnnouncements(allAnnouncements: StateAnnouncementMap): void {
     const newAnnouncements: StateAnnouncementMap = {};
-    const { allAnnouncements } = this.config;
-    Object.values(allAnnouncements).forEach(
-      (announcement: StateAnnouncement) => {
-        newAnnouncements[announcement.id] = this.state.announcements[
-          announcement.id
-        ]
-          ? this.state.announcements[announcement.id]
-          : {
-              ...announcement,
-              isShown: false,
-            };
-      },
-    );
-    this.update({ announcements: newAnnouncements });
+    this.update(({ announcements }) => {
+      Object.values(allAnnouncements).forEach(
+        (announcement: StateAnnouncement) => {
+          newAnnouncements[announcement.id] = announcements[
+            announcement.id
+          ] || { ...announcement, isShown: false };
+        },
+      );
+      announcements = newAnnouncements;
```

Comment:
I'm not sure that assigning a new variable at the end of `this.update` will have an effect. The way that `this.update` works is that it'll make a copy of the state, and then every change you make to the copy will be written back to the original state. In other words, if we want to change the state, we don't need to make a new version of the state and then replace the current state with the new version. That happens automatically! All we need to do is directly change the state. (We use Immer for handling state changes, and you can read more about the mechanics [here](https://immerjs.github.io/immer/#a-quick-example-for-comparison).)